### ISSUE TITLE
shutting down Windows when a multipass instance is running leaves instance in a broken state on next boot

### ISSUE THREAD

**Issue by ColinIanKing** [canonical/multipass#708]
I had a multipass instance running on my Windows 10 box and shut the box down accidentally. On the next reboot the instance was showing as "RUNNING" from the 'multipass list' output but I could not actually shell into the install or stop the instance either.  The multipass shell command just returns a timeout error message "ssh connection failed: 'Timeout connecting to bionic.mshome.net'.

**Comment by ColinIanKing** [comment]
I found a workaround for my instanced named 'bionic':

multipass suspend bionic
multipass stop bionic
multipass start bionic

Seems like suspending it does the trick.

**Comment by ricab** [comment]
Hi, thanks for the report. The multipass daemon should normally (if it managed to exit cleanly) have suspended the instance and brought it back up when next running. It could have failed to do so if it was forcefully killed by the OS. Is there any indication to that in the event viewer? Or clues about other failures?

**Comment by Saviq** [comment]
Hey @ColinIanKing, by "accidentally shut down", do you mean yanked the power cord, or a clean shutdown? It looks like the instance state got out of sync - it should have suspended on shutdown and resumed on power on.

**Comment by ColinIanKing** [comment]
I shut the machine down using the normal shutdown menu option as a clean shutdown, I accidentally didn't turn off the multipass instances.

**Comment by Saviq** [comment]
Right, that's a supported workflow, you should not need to shut them down.

**Comment by ricab** [comment]
Just noticed what the most likely reason is, while browsing code: we don't shutdown or suspend instances that are not running, but they can be in a few other states: starting, restarting, suspending, delayed shutdown, and unknown.

**Comment by ricab** [comment]
One way to *kind of reproduce* this is:

1. open Services on windows
2. edit Multipass Service properties
    1. startup type: change to "manual"
    2. recovery: change to "take no action" in all three dropdowns
3. use the GUI to start an instance
4. use the GUI to wait until the instance is marked as starting
5. immediately stop the daemon service (while the instance is starting)
    1. in my experience, this hangs, daemon says "Goodbye!" but doesn't quit
    2. kill the daemon process
    1. observe instance still running in Hyper-V manager

So the hang in 5.1 may be another cause. 

FWIW, "delayed shutdown" is not a reproducer because our `DelayedShutdownTimer` objects are declared after instances in `Daemon`. Because of reverse destruction order, they're deleted first, and their destructor cancels the shutdown, leaving the instance running.

**Comment by ricab** [comment]
> FWIW, "delayed shutdown" is not a reproducer because our DelayedShutdownTimer objects are declared after instances in Daemon.

VM classes should not depend on that though. They should still handle all states in the destructor.

I am unassigning myself after all, because dealing with this properly in all backends requires a few steps:

- not hanging on start + dtor
- ability to power of VMs by "cable yank" (#1909)
- clarifying VM class interface for "acpi shutdown" vs "cable yank"
- revising VM dtors across backends (w/ timeouts, watchdogs, and fallbacks)

**Comment by Sploder12** [comment]
I could not reproduce and the code has change significantly, this is likely to not be an issue anymore.
